homeassistant/components/samsungtv/bridge.py

```diff
@@ -0,0 +1,241 @@
+                LOGGER.debug("Working but unsupported config: %s", config)
+                return RESULT_NOT_SUPPORTED
+            except OSError as err:
+                LOGGER.debug("Failing config: %s, error: %s", config, err)
+
+        return RESULT_NOT_SUCCESSFUL
+
+    def get_state(self):
+        """Get TV state."""
+        if self._remote is not None:
+            # Close the current remote connection
+            self._remote.close()
+            self._remote = None
+
+        try:
+            self._get_remote()
+            if self._remote:
+                return STATE_ON
+        except (
+            UnhandledResponse,
+            AccessDenied,
+        ):
+            # We got a response so it's working.
+            return STATE_ON
+        except OSError:
+            # Different reasons, e.g. hostname not resolveable
+            return STATE_OFF
+
+        return STATE_OFF
+
+    def _get_remote(self):
+        """Create or return a remote control instance."""
+        if self._remote is None:
+            # We need to create a new instance to reconnect.
+            try:
+                LOGGER.debug("Create SamsungRemote")
+                self._remote = Remote(self.config.copy())
+            # This is only happening when the auth was switched to DENY
+            # A removed auth will lead to socket timeout because waiting for auth popup is just an open socket
+            except AccessDenied:
+                self._notify_callback()
+                raise
+        return self._remote
+
+    def _send_key(self, key):
+        """Send the key using legacy protocol."""
+        self._get_remote().control(key)
+
+
+class SamsungTVWSBridge(SamsungTVBridge):
+    """The Bridge for WebSocket TVs."""
+
+    def __init__(self, config):
+        """Initialize Bridge."""
+        super().__init__(config)
+        self.config = config
+
+    def try_connect(self, port):
+        """Try to connect to the Websocket TV."""
+        for self.port in (8001, 8002):
+            if port is None or port == self.port:
+                config = {
+                    "name": "HomeAssistant",
+                    "description": "HomeAssistant",
+                    "host": self.host,
+                    "method": self.method,
+                    "port": self.port,
+                    # We need this high timeout because waiting for auth popup is just an open socket
+                    "timeout": 31,
+                    "token": self.token,
+                }
+                try:
+                    LOGGER.debug("Try config: %s", config)
+                    with SamsungTVWS(
+                        host=self.host,
+                        port=self.port,
+                        token=self.token,
+                        timeout=config["timeout"],
+                        name=config["name"],
+                    ) as remote:
+                        remote.open()
+                    LOGGER.debug("Working config: %s", config)
```

Comment:
Please don't log the whole config with the token.

Reply:
That's only debug information... I don't see any security issue there. If you prefer I can log IF there is some token and not token value itself